### PR TITLE
Security Documentation

### PR DESCRIPTION
**Pull request by LukasRos**
This is security documentation based upon the latest Stoplight Studio version and OpenAPI specification, partly based upon [this blog article](https://stoplight.io/blog/api-security-in-openapi-b105ae725a08/).

### CODE REVIEW

File: docs/Design-and-Modeling/10-api-security.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+### OAuth 2
+
+For OAuth 2, select the _oauth2_ scheme type. OAuth is an open authentication standard defined by [RFC6749](https://tools.ietf.org/html/rfc6749) and [RFC6750](https://tools.ietf.org/html/rfc6750). OAuth also uses the standard HTTP _Authorization_ header to send a credential known as bearer token but also defines various flows for retrieving this bearer token. You can enter the configuration for the flows into your OpenAPI specification, and various OpenAPI tools can use this information to assist the user in going through them.
+
+Hint: If you are documenting an API that still uses OAuth 1, select the _http_ scheme type and then _oauth_ as a subtype instead.
+
+![OAuth Configuration](../../assets/images/security-oauth-a.png)
+
+Once selecting _oauth2_, you will see a flows input field into which you can enter one or more of the following flow types: _implicit_, _password_, _clientCredentials_, and _authorizationCode_. An autocomplete dropdown appears as soon as you click in the flows field.
```
Comment (mnaumanali94) [inline docs/Design-and-Modeling/10-api-security.md:54]: Maybe we can build upon this by talking a little about when to use different flows? Taking things from here: https://auth0.com/docs/api-auth/which-oauth-flow-to-use

Also maybe we can move OAuth 2 before HTTP?

Comment (mnaumanali94) [reply docs/Design-and-Modeling/10-api-security.md:54]: @philsturgeon What do you think?

File: docs/Design-and-Modeling/10-api-security.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+### HTTP Authentication
+
+For HTTP authentication, select the _http_ scheme type. HTTP authentication uses the standard HTTP _Authorization_ header defined in [RFC7235](https://tools.ietf.org/html/rfc7235).
+
+![HTTP Auth Configuration](../../assets/images/security-http.png)
+
+Use the right dropdown to switch between the different HTTP authentication schemes. OpenAPI and, by extension, Stoplight support all the [schemes registered at IANA](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml). Further configuration is neither necessary nor possible.
+
+While OpenAPI supports the different variants of HTTP authentication, the majority of APIs use _basic_. With basic authentication, an API client sends the combination of a username and a password in the Authorization header. The method for retrieving the username and password is not part of the OpenAPI description.
```
Comment (mnaumanali94) [inline docs/Design-and-Modeling/10-api-security.md:44]: Maybe tell them to add the method to retrieve these credentials in a quickstart or auth documentation. We can do a small section about this "Writing documentation"?

Comment (LukasRos) [reply docs/Design-and-Modeling/10-api-security.md]: Added this section, please check.

File: docs/Design-and-Modeling/10-api-security.md (hunk 1 above)
Comment (LukasRos) [reply docs/Design-and-Modeling/10-api-security.md:44]: Changed the order and included a very brief description of the flows.

**Pull request merged**